packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -225,6 +229,27 @@ class SliverConstraints extends Constraints {
   /// contents depends on the [growthDirection].
   final double scrollOffset;
 
+  /// The scroll distance that has been consumed by all [Sliver]s that came
+  /// before this [Sliver].
+  ///
+  /// [Sliver]s often lazily create their internal content as layout occurs,
+  /// e.g., [SliverList]. In this case, when [Sliver]s exceed the viewport,
+  /// their children are built lazily, and the [Sliver] does not have enough
+  /// information to estimate its total extent, [consumedScrollExtent] will be
+  /// [double.infinity] for all [Sliver]s that appear after the lazily
+  /// constructed child. This is because a total [scrollExtent] cannot be
+  /// calculated unless all inner children have been created and sized, or the
+  /// number of children and estimated extents are provided. The infinite
+  /// [scrollExtent] will become finite as soon as enough information is
+  /// available to estimate the overall extent of all children within the given
+  /// [Sliver].
+  ///
+  /// [Sliver]s may legitimately be infinite, meaning that they can scroll
+  /// content forever without reaching the end. For any [Sliver]s that appear
+  /// after the infinite [Sliver], the [consumedScrollExtent] will be
+  /// [double.infinity].
+  final double consumedScrollExtent;
```

Comment:
precedingScrollExtent? Consumed sounds more like it has something to do with the current sliver which it doesn't

Reply:
done